### ISSUE TITLE
Smelting and growing when you're not around

### ISSUE THREAD

**Issue by PianoPlayingHusky** [drewhoener/JWSuggestions#135]
IGN: CaptainSparrow_

You know anything in furnaces, brewing tables or farming crops doesn't grow when you're around? I think it should change so it still grows/smelts/brews when you're away. 
Just a little something I would like to add.

**Comment by drewhoener** [comment]
I don't think I can control that, unless I cancelled a ChunkUnloadEvent or something, but that isn't great for the server in terms of lag, and it can get dangerous with tons of TileEntities

**Comment by 5d100** [comment]
i was about to say that. the reason it doesn't smelt when you're away is lag. now, if donators got a command, that'd solve your problem at least partly with a furnace.

**Comment by drewhoener** [comment]
no, the reason it doesn't smelt when you're away is becuase that's how MC works, it unloads the chunks and the Tile Entities stop ticking. When I prevent the chunks from unloading (like they would normally), THAT'S when the lag starts

**Comment by TemperrroniPizza** [comment]
When you set a server render distance, the server is loading chunks when youre around, and saves the progress by processing tiledrops, and entities for you. But if you keep every chunk loaded, you'd need a multi ten-thousand dollar pc to run it. Soemthign like that could run you easily 12-15 g's if youre lucky. Johnnywoof doesnt have the money to afford it.

**Comment by TemperrroniPizza** [comment]
Every tile entity and causes lag, in small increments, but exponentially multiplies itself, when another entity is around. even my 12 core xeon rig lags when 400-500 entites are present. The server can run it, but it would lag like crazy. That's why unloading chunks in minecraft is so important, it saves ram usage, and doesn't cap out and crash.

**Comment by TemperrroniPizza** [comment]
The server map takes up about 512-640gb of ram of any kind loaded everywhere. With an average of 50 entities per chunk loaded, the server has to take a toll to run continuously. That's why the server reboots often, to keep ram usage low, and processing speeds snappy.

**Comment by TemperrroniPizza** [comment]
I tried this with last map, and mcedit to make the map test. And spent a few grand on ddr3 ecc ram to run this. thats the result I got, to render the whole map, it took 5 days, 7 hours, 12 minutes and 25 seconds to load.

**Comment by drewhoener** [comment]
This is grossly inaccurate and exaggerated. You're blowing way out of proportion the costs of running a server as well as how Minecraft mechanics work. 400 entities is nothing to our server (and hopefully not a Xeon that costs $1400 that I doubt you have). Also it's not just about the render distance, but also tick costs and other factors not under our control. Please keep your responses reasonable and truthful

**Comment by 5d100** [comment]
I did not understand that for the most part. thanks for the info.

**Comment by drewhoener** [comment]
@5d100 he deleted his comments, or someone did (I didn't), wasn't directed at you

**Comment by PianoPlayingHusky** [comment]
Who is Nerd in Plugin Coding?

**Comment by drewhoener** [comment]
I have screenshots of their comments, I'll post them later when I'm not so tired

**Comment by drewhoener** [comment]
![](http://i.imgur.com/yKlrRYt.jpg)
![](http://i.imgur.com/cZNHdNN.jpg)

**Comment by drewhoener** [comment]
let's not be ignorant, guys

**Comment by rockerjman222** [comment]
@TemperrroniPizza

**Comment by TemperrroniPizza** [comment]
500 entities is nothing, if that's a combined number of all the entities rendered at the current moment in the current time, wherever you pick. Picture every single chunk, every 16x16x16 chunk, with 500 entities per 16x16x16 chunk. Now picture all those chunks rendered in, and rendering every entity present in that chunk, may it be an item frame, or a mob. Then the notorious thing we'll all hate, presents itself. Of course to limit load, you can do one or both of the most efficient ways to avoid server lag. One way is to slow down the tick speed. Or two, by unloading unused chunks, that do not interfere with a player's render distance. I benchmarked this with a few trash parts, and I had made a typo. It's not 640gb of ram, it's 640mb of ram, allocated to the server, running a Conroe Intel Celeron. And it's was running ddr2 memory. Of course it lagged because not only is the CPU a single core clocked under 1.5ghz, but 640 my of ddr2 gets used up quickly. I ran a server and gathered that data. I made a few typing boo boos. I'm sorry, I should have proofread.

**Comment by TemperrroniPizza** [comment]
I apologize in advance. 400 entities combined is really nothing to any server. I was exaggerating the fact just to make a joke out of myself because I was threatening myself, and was dealing with terrible mood swings.

**Comment by TemperrroniPizza** [comment]
That's the explanation. @drewhoener @rockerjman222 ^^^^^

**Comment by TemperrroniPizza** [comment]
It seems cheesy, but that was actually an attempt that was caught before something would go terribly wrong. Thanks for broadcasting my ignorance to everyone, so everyone can see how ignorant I am, and have bad impressions of me before I even meet them.

**Comment by TemperrroniPizza** [comment]
I'm not trying to be mean, or rude, but I'm a little salted like saltine crackers, which reminds me I might need some, after you did what you did Red. I'll agree with the fact it was grossly way out of proportion. In my moody, depressing, suicidal emotional swings, I am probably the largest A-Hole on the planet. I apologize to everyone that has seen this already, as this was never the best place to post this, that's why I deleted them.

**Comment by PianoPlayingHusky** [comment]
There are bigger A-Holes than you.
